packages/template-retail-react-app/app/components/unavailable-product-confirmation-modal/index.jsx

```diff
@@ -14,28 +14,52 @@ import {useDisclosure} from '@salesforce/retail-react-app/app/components/shared/
 import {noop} from '@salesforce/retail-react-app/app/utils/utils'
 
 /**
- * This Component will responsible to determine of a given product ids has become unavailable
- * and will prompt the users to remove them before proceeding any further
+ * This Component determines if the provided products have become unavailable or out of stock or low stock that
+ * can't be fulfilled and will prompt the users to remove them before proceeding any further
  *
  * @param productIds -  list of product ids to check for availability
+ * @param productItems -  basket product items
  * @param handleUnavailableProducts - callback function to handle what to do with unavailable products
  * @returns {JSX.Element} -  Conformation Modal Component
  *
  */
 const UnavailableProductConfirmationModal = ({
     productIds = [],
+    productItems = [],
     handleUnavailableProducts = noop
 }) => {
     const unavailableProductIdsRef = useRef(null)
+    const ids = productIds.length ? productIds : productItems.map((i) => i.productId)
     useProducts(
-        {parameters: {ids: productIds?.join(','), allImages: true}},
+        {parameters: {ids: ids?.join(','), allImages: true}},
         {
-            enabled: productIds?.length > 0,
+            enabled: ids?.length > 0,
             onSuccess: (result) => {
-                // when a product is unavailable, the getProducts will not return its product detail.
-                // we compare the response ids with the ones in basket to figure which product has become unavailable
-                const resProductIds = result.data?.map((i) => i.id) || []
-                const unavailableProductIds = productIds.filter((id) => !resProductIds.includes(id))
+                const resProductIds = []
+                const unOrderableIds = []
+                result.data?.forEach(({id, inventory}) => {
+                    const productItem = productItems.find((item) => item.productId === id)
+                    // wishlist item will have the property type
+                    const isWishlist = !!productItem?.type
+                    // when a product is unavailable, the getProducts will not return its product detail.
+                    // we compare the response ids with the ones in basket to figure which product has become unavailable
+                    resProductIds.push(id)
+                    // when a product is orderable, but the quantity in the basket is more than the remaining stock
+                    // we want to make sure it is removed before go to checkout page to avoid error when placing order
+                    // we don't need to remove low stock/ out of stock from wishlist
+                    if (
+                        !isWishlist &&
+                        (!inventory?.orderable ||
+                            (inventory?.orderable && productItem?.quantity > inventory.stockLevel))
+                    ) {
+                        unOrderableIds.push(id)
+                    }
```

Comment:
It looks like the logic here still assumes that `productItems` prop is passed in. Can we tweak the logic like this?

- If `productIds` props is passed in, logic will check for `resProductIds` only.
- If `productItems` is passed in, the logic will check for both `resProductIds` and `unOrderableIds`.